src/main/java/core/basesyntax/service/impl/CsvFormatReportGenerator.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.service.ReportGenerator;
+import java.util.stream.Collectors;
+
+public class CsvFormatReportGenerator implements ReportGenerator {
+    private static final String HEADER = "fruit,quantity";
+    private static final String KEY_VALUE_SEPARATOR = ",";
+    private static final String LINE_SEPARATOR = "\n";
+    private final FruitStorageDao fruitStorageDao;
+
+    public CsvFormatReportGenerator(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public String getReport() {
+        String reportData = fruitStorageDao.getAllFruits().stream()
```

Comment:
It appears that the `fruit.toString()` method is implicitly called in the stream operation. Ensure that the `Fruit` class overrides the `toString()` method to return the correct representation of the fruit name for the report. If not, this will lead to an incorrect report format.